javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -53,13 +48,13 @@
         Objects.requireNonNull(classifier, "classifier is null");
         Objects.requireNonNull(mapper, "mapper is null");
 
-        final java.util.Map<C, Collection<T>> mutableResults = new java.util.HashMap<>();
+        final java.util.Map<C, Collection<T>> mutableResults = new java.util.LinkedHashMap<>();
```

Comment:
@danieldietrich, @ruslansennov, this is where the magic happens, i.e. all `groupBy`s using a `LinkedHashMap` instead of a `HashMap`